multinet/api/tests/test_network.py

```diff
@@ -430,15 +432,19 @@ def test_network_rest_retrieve_tables_type(
     response = authenticated_api_client.get(
         f'/api/workspaces/{workspace.name}/networks/{network.name}/tables/', data={'type': type}
     )
-    assert response.status_code == 200
-
-    if type == 'node':
-        assert len(response.data) == len(node_tables)
-        for table in response.data:
-            assert not table['edge']
-            assert table['name'] in node_tables
-    else:  # type = 'edge'
-        assert len(response.data) == 1  # test network created with one edge definition
-        for table in response.data:
-            assert table['edge']
-            assert table['name'] == edge_table.name
+    if success:
+        assert response.status_code == 200
+        if type == 'node':
+            assert len(response.data) == len(node_tables)
+            for table in response.data:
+                assert not table['edge']
+                assert table['name'] in node_tables
+        elif type == 'edge':  # type = 'edge'
+            assert len(response.data) == 1  # test network created with one edge definition
+            for table in response.data:
+                assert table['edge']
+                assert table['name'] == edge_table.name
+        else:  # type == 'all'
+            assert {*node_tables, edge_table.name} == {table['name'] for table in response.data}
```

Comment:
I think this ought to be an `elif` block, with an `else` below to catch the case where the test has received a bad parameter.

Reply:
This function represents four test cases: three for successful calls (node, edge, all), and one for an unsuccessful call (foo). As written, in the `if success:` block, `type` will only be one of (node, edge, all), and the `else` block corresponding to the `if success:` catches a bad parameter.

I can see how adding new test cases under the `pytest.mark.parametrize` decorator might muddy this distinction in the future though, so if either you or @AlmightyYakob feel strongly about me rewriting this now I will.

Reply:
I think the way this is is fine since the general `if success: else:` pattern is common within our tests. It just so happens that there's only one unsuccessful case in this test, which is directly tied to the `type` parameter.

Reply:
Let's stick with this then, but in general I believe it's better to be more explicit when writing these sort of "switch statements". The giveaway to me is that we need a comment to explain the logic of expecting `type == 'all'` in the else block. My general preference would be to make the `all` case explicit with an elif, then have an else block unconditionally throw a runtime error indicating that indeed something went very, very wrong.